cm-vault/vault-config/main.tf

```diff
@@ -42,3 +42,45 @@ resource "vault_pki_secret_backend_root_cert" "ca" {
   key_bits             = 2048
   exclude_cn_from_sans = true
 }
+
+# AppRole Authentication Method
+resource "vault_auth_backend" "approle" {
+  count = var.vault_authentication_mechanism == "approle" ? 1 : 0
+
+  type = "approle"
+}
+
+resource "vault_approle_auth_backend_role" "auth_role" {
+  count = var.vault_authentication_mechanism == "approle" ? 1 : 0
+
+  backend        = vault_auth_backend.approle[0].path
+  role_name      = "auth-role"
+  token_policies = ["default", "dev", "prod"]
```

Comment:
Suggestion: I think we only have one policy (created on line 68).
I think it would be better to refer to the name of that policy via the resource i.e `token_policies = [vault_policy.approle-pki-policy[count.index].name]` as that way the relationship between the two resources is made clear.